src/coral_models/prepare_raw_data.py

```diff
@@ -0,0 +1,470 @@
+        cnx = sqlite3.connect(read_aloud_path / "db.sqlite3")
+        read_aloud_data = pd.read_sql_query("SELECT * FROM CoRal_recording", cnx)
+        read_aloud_data_speakers = read_aloud_data[db_to_excel_metadata_names.keys()]
+        read_aloud_data_speakers = read_aloud_data_speakers.rename(
+            columns=db_to_excel_metadata_names
+        )
+        speaker_list.append(read_aloud_data_speakers)
+
+    # Concatenate all speaker information
+    speakers = pd.concat(speaker_list, axis=0).drop_duplicates().reset_index(drop=True)
+
+    # People often typed their name in all lower case or similiar variations, which
+    # yields a lot of duplicates. We therefore we remove rows with dublicate emails.
+    speakers = speakers.drop_duplicates(subset=["mail"])
+
+    # The native_language columns contains both aplha-2 codes and full names, so we
+    # need to convert the aplha-2 codes to full names. The same goes for the
+    # birthplace. Spoken_languages is a list of languages, so we need to convert that
+    # as well.
+    speakers["native_language"] = speakers["native_language"].apply(correct_language)
+    speakers["birthplace"] = speakers["birthplace"].apply(correct_language)
+    speakers["spoken_languages"] = speakers["spoken_languages"].apply(
+        correct_country_list
+    )
+
+    # Specification of gender is not consistent, so we need to correct that
+    # by mapping "K" to "female", and "M" to "male".
+    speakers["gender"] = speakers["gender"].apply(
+        lambda x: "male"
+        if x in ["male", "M"]
+        else "female"
+        if x in ["K", "female"]
+        else x
+    )
+
+    # Create a speaker id column
+    speakers["speaker_id"] = "t" + speakers.index.astype(str)
+
+    return speakers
+
+
+def make_recording_metadata(
+    cfg: DictConfig, speakers: pd.DataFrame, raw_path: Path, sentences: pd.DataFrame
+) -> pd.DataFrame:
+    """Make a recording metadata file from the raw data
+
+    Args:
+        cfg (DictConfig):
+            The config
+        speakers (pd.DataFrame):
+            The speakers dataframe
+        raw_path (Path):
+            The path to the raw data
+        sentences (pd.DataFrame):
+            The sentences dataframe
+
+    Returns:
+        pd.DataFrame: The recording metadata dataframe"""
+    # Load the metadata
+    metadata_path = raw_path / "metadata.xlsx"
+
+    # Make speaker-metadata file
+    speaker_metadata = pd.read_excel(metadata_path)
+
+    # Get the columns that contain information about speakers, recorders and recording
+    recording_metadata = speaker_metadata[
+        [
+            col
+            for col in speaker_metadata.columns
+            if not any(
+                [
+                    substring in col
+                    for substring in ["subject_a", "subject_b", "recorder"]
+                ]
+            )
+        ]
+        + ["subject_a_mail", "subject_b_mail", "recorder_mail"]
+    ]
```

Comment:
This seems like it gets all the columns which are **not** about the speakers and recorders, because of the `not`?